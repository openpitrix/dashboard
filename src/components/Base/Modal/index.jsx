import { omit } from 'lodash';
import React from 'react';
import ReactModal from 'react-modal';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { translate } from 'react-i18next';

import Icon from '../Icon';
import Button from '../Button';
import styles from './index.scss';

ReactModal.defaultStyles.overlay = Object.assign(
  {},
  ReactModal.defaultStyles.overlay,
  {
    padding: 0,
    backgroundColor: 'rgba(0, 0, 0, 0.7)',
    zIndex: 2000
  }
);

ReactModal.defaultStyles.content = Object.assign(
  {},
  omit(ReactModal.defaultStyles.content, [
    'top',
    'left',
    'right',
    'bottom',
    'padding'
  ]),
  {
    width: 744,
    position: 'relative',
    margin: '0 auto'
  }
);

@translate()
export default class Modal extends React.Component {
  static propTypes = {
    cancelText: PropTypes.string,
    children: PropTypes.any,
    className: PropTypes.string,
    closable: PropTypes.bool,
    hideFooter: PropTypes.bool,
    hideHeader: PropTypes.bool,
    isDialog: PropTypes.bool,
    okText: PropTypes.string,
    onCancel: PropTypes.func,
    onOk: PropTypes.func,
    title: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
    visible: PropTypes.bool,
    width: PropTypes.number
  };

  static defaultProps = {
    className: '',
    width: 744,
    hideHeader: false,
    visible: false,
    closable: true,
    isDialog: false,
    onOk() {},
    onCancel() {}
  };

  render() {
    const {
      className,
      title,
      width,
      visible,
      children,
      hideHeader,
      hideFooter,
      onOk,
      onCancel,
      okText,
      cancelText,
      closable,
      isDialog,
      t
    } = this.props;
    const style = { content: { width } };

    return (
      <ReactModal
        className={classnames(styles.modal, className)}
        style={style}
        isOpen={visible}
        onRequestClose={onCancel}
        ariaHideApp={false}
      >
        {!hideHeader && (
          <div className={styles.header}>
            <div className={styles.title}>{title}</div>
            {closable && <Icon name="close" size={36} onClick={onCancel} />}
          </div>
        )}
        <div className={styles.body}>{children}</div>
        {!hideFooter && (
          <div className={styles.footer}>
            <div className={classnames({ [styles.operationBtn]: !isDialog })}>
              <Button type="primary" onClick={onOk}>
                {okText || t('Confirm')}
              </Button>
              <Button type="default" onClick={onCancel}>
                {cancelText || t('Cancel')}
              </Button>
            </div>
          </div>
        )}
      </ReactModal>
    );
  }
}
