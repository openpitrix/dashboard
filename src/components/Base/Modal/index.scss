@import '~scss/vars';

.modal {
  position: relative;
  width: 744px;
  max-height: 600px;
  margin: 0 auto;
  margin-top: 100px;
  border-radius: 3px;
  background-color: $N0;
  outline: none;
  overflow: auto;

  .header {
    position: relative;
    padding: 20px 32px;
    border-bottom: 1px solid $N10;
    .title {
      font-size: 20px;
      line-height: 1.2;
      font-weight: 500;
      color: $N500;
    }
    :global {
      .icon-close {
        position: absolute;
        top: 32px;
        right: 22px;
        cursor: pointer;
      }
    }
  }

  .body {
    padding: 20px 32px 30px;
    overflow-y: auto;
    :global{
      .formContent{
        margin: 12px 0 -18px;
        & > div, .inputItem{
          margin-bottom: 24px;
          & > label{
            display: inline-block;
            margin-right:  24px;
            width: 168px;
            font-size: 14px;
            font-weight: 500;
            line-height: 1.14;
            text-align: right;
            color: $N500;
          }
          & > input{
            width: 256px;
            height: 32px;
            border-radius: 2px;
            background-color: $N0;
            box-shadow: 0 1px 2px 0 rgba(109, 53, 195, 0.15);
            border: solid 1px $P75;
          }
          & > textarea{
            padding: 7px 10px;
            width: 400px;
            height: 100px;
            border-radius: 2px;
            border: 1px solid #e7e7e7;
            outline: none;
            resize: none;
            border-radius: 2px;
            background-color: $N0;
            border: solid 1px $N30;
            &:focus{
              border: 1px solid $border-dark;
            }
          }
          & > select{
            width: 256px;
            height: 32px;
            border-radius: 2px;
            background-color: $N0;
            box-shadow: 0 1px 2px 0 rgba(109, 53, 195, 0.15);
            border: solid 1px $P75;
          }
        }
        .textareaItem{
          height: 100px;
          & > label {
            position: relative;
            bottom: 80px;
          }
        }
        .selectItem{
          & > label{
            float: left;
            line-height: 32px;
          }
        }
      }
    }
  }
  .footer {
    padding: 16px 32px;
    border-top: 1px solid $N10;
    text-align: right;
    .operationBtn{
      margin-left: 192px;
      text-align: left;
    }
    button {
      margin-right: 12px;
      &:last-of-type {
        margin-right: 0;
      }
    }

  }
}


