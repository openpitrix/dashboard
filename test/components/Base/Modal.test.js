import Modal from 'components/Base/Modal';

describe('Base/Modal', () => {
  it('basic render', () => {
    const wrapper = <Modal>test2</Modal>;

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
