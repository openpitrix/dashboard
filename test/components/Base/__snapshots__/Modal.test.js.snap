// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base/Modal basic render 1`] = `undefined`;
